core/base/src/TListOfTypes.cxx

```diff
@@ -63,28 +63,19 @@ static bool NameExistsElsewhere(const char* name){
       // We have a scope
       const auto enName = lastPos + 1;
       const auto scopeNameSize = ((Long64_t)lastPos - (Long64_t)name) / sizeof(decltype(*lastPos)) - 1;
-#ifdef R__WIN32
-      char *scopeName = new char[scopeNameSize + 1];
-#else
-      char scopeName[scopeNameSize + 1]; // on the stack, +1 for the terminating character '\0'
-#endif
-      strncpy(scopeName, name, scopeNameSize);
-      scopeName[scopeNameSize] = '\0';
+      std::string scopeName{name, scopeNameSize};
```

Comment:
```suggestion
      TString scopeName{name, scopeNameSize};
```
is equivalent and 'avoids' the need for the `.c_str()` changes 

(On a separate note, ideally we would actually use `std::string_view` but to be useful this would require the called interface to have an implementation that leverage the parameter being a string view)

Reply:
I don't think there is any reason to use non-standard C++ classes in this context.